core/src/main/scala/cats/syntax/all.scala

```diff
@@ -40,6 +40,7 @@ trait AllSyntax
     with TransLiftSyntax
     with TraverseFilterSyntax
     with TraverseSyntax
+    with TupleSyntax
```

Comment:
This needs to also have `object tuple extends TupleSyntax` in `package.scala` right?

Also, do we want to put this under tuples or Cartesian syntax?